src/main/java/io/debezium/connector/jdbc/JdbcSinkConnectorTask.java

```diff
@@ -0,0 +1,164 @@
+
+import javax.persistence.EntityManager;
+import javax.xml.bind.JAXBException;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.apache.kafka.connect.sink.SinkTask;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.boot.jaxb.hbm.spi.JaxbHbmHibernateMapping;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.data.Envelope;
+
+/**
+ * The main task executing streaming from sink connector.
+ * Responsible for lifecycle management of the streaming code.
+ *
+ * @author Hossein Torabi
+ *
+ */
+public class JdbcSinkConnectorTask extends SinkTask {
+    private static enum State {
+        RUNNING,
+        STOPPED;
+    }
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JdbcSinkConnectorTask.class);
+
+    private final AtomicReference<State> state = new AtomicReference<State>(State.STOPPED);
+    private final ReentrantLock stateLock = new ReentrantLock();
+    private final Map<String, Schema> topicSchema = new HashMap<>();
+
+    private JdbcSinkConnectorConfig config;
+
+    private SessionFactory sessionFactory;
+    private EntityManager entityManager;
+    private Session session;
+
+    @Override
+    public String version() {
+        return Module.version();
+    }
+
+    @Override
+    public void start(Map<String, String> props) {
+        stateLock.lock();
+
+        try {
+            if (!state.compareAndSet(State.STOPPED, State.RUNNING)) {
+                LOGGER.info("Connector has already been started");
+                return;
+            }
+
+            config = new JdbcSinkConnectorConfig(props);
+            config.validate();
+
+            buildSessionFactory();
+
+        }
+        finally {
+            stateLock.unlock();
+        }
+    }
+
+    @Override
+    public void put(Collection<SinkRecord> records) {
+        try {
+            session = entityManager.unwrap(Session.class);
+            Transaction transaction = session.beginTransaction();
+            for (SinkRecord record : records) {
+                Struct value = (Struct) record.value();
+                Schema valueSchema = record.valueSchema();
+                String topic = record.topic();
+                if (!Envelope.isEnvelopeSchema(valueSchema)) {
+                    throw new RuntimeException("Only Debezium schema is supported");
+                }
+
+                Struct afterStruct = value.getStruct(Envelope.FieldName.AFTER);
+                Schema afterSchema = afterStruct.schema();
+
+                if (!topicSchema.containsKey(topic) || !topicSchema.get(topic).equals(afterSchema)) {
+                    topicSchema.put(topic, afterSchema);
+                    JaxbHbmHibernateMapping mapping = HibernateJaxbRecordMapper.getJaxbHbmHibernateMapping(topic, afterSchema);
```

Comment:
Is there a specific reason why we need a JAXB XML in the first place?  You are effectively using map mode later so I'm struggling to grok the purpose of this other than trying to detect when a topic's schema has changed.  If that's true, couldn't we resolve topic schema deltas without this by keeping some in-memory state of the table's structure and comparing that to the incoming event's structure?

The implementation appears to rely on creating a JAXB XML, writing it to disk, and loading it.  As we found with the Oracle connector's embedded ISPN support, this puts a requirement that the connector must have a network shared drive mounted for this to work and is tough to set-up in some environments.

Reply:
Actually initial idea of this connector was generate Jaxb mapping file from record's schema and use map to persist data.
So there is nu argue about that and we can have changes

Reply:
There are two ways in which you can interact with Hibernate.

The first is where you need to supply an entity mapping, whether that is via JAXB, `orm.xml`, or an annotated entity class.  This is where Hibernate resolves the entity persistence model from the mapping.  The second is where you explicitly configure the `SessionFactory` to operate in what is called map-mode.  In this mode there isn't a need for mappings at all, Hibernate resolves the mapping on the fly by resolving Java type to SQL types.

If you use dynamic map mode, there won't be any need for JAXB and therefore there shouldn't be a need to restart the Hibernate `SessionFactory` when you encounter a new entity type in the sink record stream iiuc.